src/lib/download-manager.js

```diff
@@ -96,6 +97,53 @@ class DownloadManager {
 		}
 	}
 
+	async _validateCacheLimit({ url, maxCacheSize = settings.tachyonCacheLimit, currentFileName }) {
+		const fileHeaders = await fetch(url, { method: 'HEAD' });
+
+		const contentLength = parseInt(fileHeaders.headers.get('content-length') || '0', 10);
+		// get the size of the download directory
+		const downloadDirStats = await this.getDownloadedFilesStats(currentFileName);
+		const totalSize = (downloadDirStats.totalSize + contentLength) / (1024 * 1024 * 1024); // convert to GB
+		if (totalSize > maxCacheSize && downloadDirStats.fileStats.length > 0) {
+			// prompt to delete files
+			const question = {
+				type: 'confirm',
+				name: 'cleanCache',
+				message: 'Downloaded files exceed the cache limit. Do you want to clean up the cache?',
+				default: true
+			};
+			const answer = await this.ui.prompt(question);
+			if (answer.cleanCache) {
+				await this._freeCacheSpace(downloadDirStats);
+			} else {
+				// remove files until the cache limit is satisfied
+				this.ui.write('Cache cleanup skipped. Remove files manually to free up space.');
+			}
+		}
+		return;
+	}
+
+	async getDownloadedFilesStats(currentFile) {
```

Comment:
This function can be improved a bit: filter out files that are not normal files (like directories), filter out files that don't end with .zip or .progress in case there's other stuff in there, only call stat once on each file.